internal/integration/setup.go

```diff
@@ -0,0 +1,49 @@
+package integration
+
+import (
+	"context"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	LogOccurance          = 2
+	ContainerStartTimeout = 10 * time.Second
+)
+
+func createTestDBContainer() (*postgres.PostgresContainer, error) {
+	ctx := context.Background()
+
+	dbName := "test"
+	dbUser := "testuser"
+	dbPassword := "postgres"
+
+	postgresContainer, err := postgres.RunContainer(ctx,
+		testcontainers.WithImage("docker.io/postgres:16-alpine"),
+		postgres.WithDatabase(dbName),
+		postgres.WithUsername(dbUser),
+		postgres.WithPassword(dbPassword),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("database system is ready to accept connections").
+				WithOccurrence(LogOccurance).
+				WithStartupTimeout(ContainerStartTimeout)),
+	)
+	if err != nil {
+		return nil, err
+	}
+	_, _, err = postgresContainer.Exec(ctx, []string{"psql", "-U", dbUser,
+		"-d", dbName,
+		"-c", "CREATE TABLE emails (id SERIAL PRIMARY KEY, email TEXT UNIQUE NOT NULL)"})
```

Comment:
If table structure will change you have to add a migration script and fix script in test.
It's better to unify this process and make test container to apply the same migrations that app does when starting

Reply:
Absolutely agree. This workaround is going to create more problems in the future. It's temporary solution, will change it with next commit.

Reply:
Resolved with [dd0b305](https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-fdemchenko/pull/4/commits/dd0b305bea1a7ac8419f040898841fdf188ce356). Now, migrations applying to test container.